spec/features/admin/legislation/proposals_spec.rb

```diff
@@ -0,0 +1,100 @@
+require 'rails_helper'
+
+feature 'Admin legislation processes' do
+
+  background do
+    admin = create(:administrator)
+    login_as(admin.user)
+  end
+
+  context "Index" do
+
+    scenario 'Displaying legislation proposals' do
+      proposal = create(:legislation_proposal, cached_votes_up: 10)
+      visit admin_legislation_process_proposals_path(proposal.legislation_process_id)
+      within "#legislation_proposal_#{proposal.id}" do
+        expect(page).to have_content(proposal.title)
+        expect(page).to have_content(proposal.id)
+        expect(page).to have_content(proposal.cached_votes_up)
+        expect(page).to have_content('Select')
+      end
+    end
+
+    scenario 'Selecting legislation proposals' do
+      proposal = create(:legislation_proposal, cached_votes_up: 10)
+      visit admin_legislation_process_proposals_path(proposal.legislation_process_id)
+
+      click_link 'Select'
+      within "#legislation_proposal_#{proposal.id}" do
+        expect(page).to have_content('Selected')
+      end
+    end
+
+    scenario 'Sorting legislation proposals by title', js: true do
+      legislation_process = create(:legislation_process)
+      create(:legislation_proposal, title: 'bbbb', cached_votes_up: 10, legislation_process_id: legislation_process.id)
+      create(:legislation_proposal, title: 'aaaa', cached_votes_up: 20, legislation_process_id: legislation_process.id)
+      create(:legislation_proposal, title: 'cccc', cached_votes_up: 30, legislation_process_id: legislation_process.id)
+      visit admin_legislation_process_proposals_path(legislation_process.id)
+
+      select "Title", from: "order-selector-participation"
+
+      within('#proposals_table') do
+        within(:xpath, "//tbody/tr[1]") do
+          expect(page).to have_content('aaaa')
+        end
+        within(:xpath, "//tbody/tr[2]") do
+          expect(page).to have_content('bbbb')
+        end
+        within(:xpath, "//tbody/tr[3]") do
+          expect(page).to have_content('cccc')
+        end
+      end
+    end
+
+    scenario 'Sorting legislation proposals by supports', js: true do
+      legislation_process = create(:legislation_process)
+      create(:legislation_proposal, title: 'bbbb', cached_votes_up: 10, legislation_process_id: legislation_process.id)
+      create(:legislation_proposal, title: 'aaaa', cached_votes_up: 20, legislation_process_id: legislation_process.id)
+      create(:legislation_proposal, title: 'cccc', cached_votes_up: 30, legislation_process_id: legislation_process.id)
+      visit admin_legislation_process_proposals_path(legislation_process.id)
+
+      select "Supports", from: "order-selector-participation"
+
+      within('#proposals_table') do
+        within(:xpath, "//tbody/tr[1]") do
+          expect(page).to have_content('10')
```

Comment:
We might want to modify this test if we decide to show records in descending order.